plugins/index.vim

```diff
@@ -2,13 +2,33 @@ python3 << EOF
 import os
 import sys
 import subprocess
+import re
 
 if "VIRTUAL_ENV" in os.environ:
+    print("Python venv detected. Attempting to update environment variables...")
     project_base_dir = os.environ["VIRTUAL_ENV"]
     script = os.path.join(project_base_dir, "bin/activate")
     pipe = subprocess.Popen(". %s; env" % script, stdout=subprocess.PIPE, shell=True)
-    output = pipe.communicate()[0].decode('utf8').splitlines()
-    env = dict((line.split("=", 1) for line in output))
-    os.environ.update(env)
+    output = pipe.communicate()[0].decode("utf8").splitlines()
 
+    def update_os_environ(line: str) -> None:
+        env_var, env_val = line.split("=", 1)
+        os.environ[env_var] = env_val
+        print("Updateing env var: " + env_var)
+
```

Comment:
Can you move this above the `if "VIRTUAL_ENV"` statement?

Reply:
Applied in second commit